crm_compassion/model/event_compassion.py

```diff
@@ -13,6 +13,62 @@
 from datetime import datetime
 
 
+class CompassionHold(models.Model):
+    _inherit = 'compassion.hold'
+
+    event_id = fields.Many2one('crm.event.compassion', 'Event', readonly=True)
+
+
+class GlobalChildSearch(models.TransientModel):
+    _inherit = 'compassion.childpool.search'
+
+    take = fields.Integer(
+        default=lambda self: self.default_number_reserve_children(),
+        readonly=lambda self: self[
+            'compassion.childpool.search'].is_from_event()
+    )
+
+    def is_from_event(self):
+        number_allocate_children = \
+            self.env.context.get('number_allocate_children')
+        return type(number_allocate_children) is int
+
+    def default_number_reserve_children(self):
+        number_allocate_children = \
+            self.env.context.get('number_allocate_children')
+        if type(number_allocate_children) is int:
+            return number_allocate_children
+        else:
+            return 80
+
+
+class ChildHoldWizard(models.TransientModel):
+    _inherit = 'child.hold.wizard'
+
+    @api.multi
+    def create_hold_vals(self, child_comp):
+        hold_vals = super(ChildHoldWizard, self).create_hold_vals(child_comp)
+
+        if self.env.context.get('event_id') is None:
+            return hold_vals
+        else:
+            hold_vals.update({'event_id': self.env.context.get('event_id')})
+            return hold_vals
+
+    @api.multi
+    def send(self):
+        dictionnary = super(ChildHoldWizard, self).send()
+
+        if self.env.context.get('event_id') is None:
+            return dictionnary
+        else:
+            del dictionnary['domain']
+            dictionnary.update({'view_mode': 'form,tree'})
+            dictionnary.update({'res_model': 'crm.event.compassion'})

```

Comment:
You can call a single update and put all your fields to update in the dictionary.

```
dictionnary.update({
    'view_mode': '...',
    'res_model': '...',
    ...
})
```
PS: you can rename dictionnary to `action`